File: Javascript/Algorithms/744.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
let nextGreatestLetter = function(letters, target) {
  if (target >= letters[letters.length - 1]) return letters[0];

  for (let c of letters) {
    if (target < c) return c;
  }
};

File: C/Algorithms/744.c
char nextGreatestLetter(char *letters, int lettersSize, char target) {
  // Handle wrap around case (larger than largest char in letters)
  // => start from beginning again.
  if (target >= letters[lettersSize - 1])
    return *letters;

  do {
    if (target < *letters)
      break;
  } while (*letters++);
  return *letters;
}